NetKet/Sampler/custom_sampler.hpp

```diff
@@ -108,31 +105,14 @@ class CustomSampler : public AbstractSampler<WfType> {
     nstates_ = hilbert_.LocalSize();
     localstates_ = hilbert_.LocalStates();
 
-    Seed();
-
     Reset(true);
 
     InfoMessage() << "Custom Metropolis sampler is ready " << std::endl;
   }
 
-  void Seed(int baseseed = 0) {
-    std::random_device rd;
-    std::vector<int> seeds(totalnodes_);
-
-    if (mynode_ == 0) {
-      for (int i = 0; i < totalnodes_; i++) {
-        seeds[i] = rd() + baseseed;
-      }
-    }
-
-    SendToAll(seeds);
-
-    rgen_.seed(seeds[mynode_]);
-  }
-
   void Reset(bool initrandom = false) override {
     if (initrandom) {
-      hilbert_.RandomVals(v_, rgen_);
+      hilbert_.RandomVals(v_, this->GetRandomEngine());
```

Comment:
maybe better just to store the random engine as a private variable? the code would look nicer, maybe marginally faster too?

Reply:
The compiler should inline the call to `GetRandomngine()` so the performance should be identical. (Compare https://godbolt.org/z/Ny9zuB for a simplified test, where the same code is generated for `B::f1` and `B::f2`.)

According to the Google Style Guide, [data members should be private](https://google.github.io/styleguide/cppguide.html#Inheritance). Making `engine_` protected would work, but I think it's more confusing than inheriting the protected function.

Reply:
Alright good points!